packages/app/src/cli/utilities/developer-platform-client/app-management-client.ts

```diff
@@ -626,29 +626,39 @@ export class AppManagementClient implements DeveloperPlatformClient {
     bundleUrl,
     skipPublish: noRelease,
   }: AppDeployOptions): Promise<AppDeploySchema> {
-    // `name` is from the package.json package name or the directory name, while
-    // the branding module reflects the current specified name in the TOML.
-    // Since it is technically valid to not have a branding module, we will default
-    // to the `name` if no branding module is present.
-    let updatedName = name
-    const brandingModule = appModules?.find((mod) => mod.specificationIdentifier === BrandingSpecIdentifier)
-    if (brandingModule) {
-      updatedName = JSON.parse(brandingModule.config).name
+    let versionInput
+    if (bundleUrl) {
```

Comment:
Even if there are no assets, we want to build the manifest and upload it to GCS no? so there will always be a bundleUrl? In which cases can we deploy without a bundle URL?

Reply:
If/when we need to store a manifest in GCS, I think it would be easy to generate one on the back end for the no asset case. In general I think it is worthwhile to have an API for creating app versions that does not require the client to make 3 network calls for a new app version (request bundle url, upload bundle, create the version).